controllers/configentry_controller_test.go

```diff
@@ -480,3 +653,149 @@ func TestConfigEntryControllers_errorUpdatesSyncStatus(t *testing.T) {
 		})
 	}
 }
+
+// Test that if the config entry hasn't changed in Consul but our resource
+// synced status isn't set to true then we update its status
+func TestConfigEntryControllers_setsSyncedToTrue(t *testing.T) {
+	t.Parallel()
+	kubeNS := "default"
+
+	cases := []struct {
+		kubeKind            string
+		consulKind          string
+		consulPrereq        capi.ConfigEntry
+		configEntryResource common.ConfigEntryResource
+		reconciler          func(client.Client, *capi.Client, logr.Logger) testReconciler
+	}{
+		{
+			kubeKind:   "ServiceDefaults",
+			consulKind: capi.ServiceDefaults,
+			configEntryResource: &v1alpha1.ServiceDefaults{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: kubeNS,
+				},
+				Spec: v1alpha1.ServiceDefaultsSpec{
+					Protocol: "http",
+				},
```

Comment:
We could explicitly set the Synced status here to `unknown`? It makes the test case feel more explicit. Also, you might need to add a case for proxydefaults